src/Microsoft.Extensions.Logging.EventSource/LoggingEventSource.cs

```diff
@@ -0,0 +1,236 @@
+    ///      
+    /// Where Name is the name of a ILoggger (case matters), Name can have a * which acts as a wildcard 
+    /// AS A SUFFIX.   Thus Net* will match any loggers that start with the 'Net'.   
+    /// 
+    /// The LEVEL is a number or a LogLevel string. 0=Trace, 1=Debug, 2=Information, 3=Warning,  4=Error, Critical=5
+    /// This speicifies the level for the associated pattern.  If the number is not specified, (first form 
+    /// of the specification) it is the default level for the EventSource.   
+    /// 
+    /// First match is used if a partciular name matches more than one pattern.   
+    ///
+    /// In addition the level and FilterSpec argument, you can also set EventSource Keywords.  See the Keywords
+    /// definition below, but basically you get to decide if you wish to have 
+    /// 
+    ///   * Keywords.Message - You get the event with the data in parsed form.
+    ///   * Keywords.JsonMessage - you get an event with the data in parse form but as a JSON blob (not broken up by argument ...)
+    ///   * Keywords.FormattedMessage - you get an event with the data formatted as a string
+    ///
+    /// It is expected that you will turn only one of these keywords on at a time, but you can turn them all on (and get
+    /// the same data logged three different ways.  
+    /// 
+    /// Example Usage 
+    /// 
+    /// This example shows how to use an EventListener to get ILogging information
+    /// 
+    /// class MyEventListener : EventListener {
+    ///     protected override void OnEventSourceCreated(EventSource eventSource) {
+    ///         if (eventSource.Name == "Microsoft-Extensions-Logging") {
+    ///             // initialize a string, string dictionary of arguments to pass to the EventSource.  
+    ///             // Turn on loggers matching App* to Information, everything else (*) is the default level (which is EventLevel.Error)
+    ///             var args = new Dictionary&lt;string, string&gt;() { { "FilterSpecs", "App*:Information;*" } };
+    ///             // Set the default level (verbosity) to Error, and only ask for the formatted messages in this case.  
+    ///             EnableEvents(eventSource, EventLevel.Error, LoggingEventSource.Keywords.FormattedMessage, args);
+    ///         }
+    ///     }
+    ///     protected override void OnEventWritten(EventWrittenEventArgs eventData) {
+    ///         // Look for the formatted message event, which has the following argument layout (as defined in the LoggingEventSource.  
+    ///         // FormattedMessage(LogLevel Level, int FactoryID, string LoggerName, string EventId, string FormattedMessage);
+    ///         if (eventData.EventName == "FormattedMessage")
+    ///             Console.WriteLine("Logger {0}: {1}", eventData.Payload[2], eventData.Payload[4]);
+    ///     }
+    /// }
+    /// </summary>
+    [EventSource(Name = "Microsoft-Extensions-Logging")]
```

Comment:
This name represents a well-known ID of the EventSource/ETW provider that exposes ILogger data to the ETW world. It will be used to subscribe to ILogger data by consumers who use EventSource/ETW to capture diagnostic traces.